Core/Src/monitor.c

```diff
@@ -51,7 +58,26 @@ void read_lv_sense(void *arg)
 	// get final voltage
 	v_int = (uint32_t)(v_dec * 10.0);
 
-	memcpy(msg.data, &v_int, msg.len);
+	// Calculate SoC using logistic function
+	// - Normal charged voltage is 29.4V
+	// - 18650 max charged voltage is 4.2V
+	// - 29.4V/4.2V = 7 batteries
+	// - Divide v_dec by 7 to get avg voltage over all rows
+	// - Use logistic function to model SoC
+	// https://www.aegisbattery.com/products/24v-20ah-li-ion-battery-pvc
+	float v_max = 4.2; // max avg voltage over all rows (7)
+	float v_min = 2.8; // min avg voltage over all rows (7)
+	float k = -8.5; // logistic fn parameter to affect steepness of curve
```

Comment:
label parts of algo `static const` if they do not change for readability sake, altough the compiler will optimize this stuff away anyways.